ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,131 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+// fakeContainerInstanceDirector imitates the ContainerInstanceDirector.
+type fakeContainerInstanceDirector struct {
+	fakeCI fakeContainerInstance
+}
+
+// fakeContainerInstance imitates the ContainerInstance.
+type fakeContainerInstance struct {
+	LatestTaskManifestSequenceNumber int64
+	ManifestMessageID                string
+}
+
+type fakeTaskComparer struct {
```

Comment:
How about using gomock to generate the mock implementation? An implementation generated by gomock will make sure the methods are called the expected number of times with the expected arguments.

Reply:
The container instance director that is spoofed here is not in `ecs-agent`. I do not know what methods we can mock by using such an approach either

Reply:
`fakeTaskComparer` doesn't seem to have anything to do with container instance director? I don't see why we cannot use mockgen to generate a mock implementation of `TaskComparer` for use in testing. A benefit of a mock generated by mockgen is that we won't have to write the mock ourselves and we get assertions on interface interactions for free.